src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -39,23 +51,55 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
         $this->joinRelations($queryBuilder, $resourceClass);
     }
 
+    public function getMetadataProperties($resourceClass): array {
+        $properties = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
+            $properties[$property] = $this->propertyMetadataFactory->create($resourceClass, $property);
+        }
+
+        return $properties;
+    }
+
     /**
      * Left joins relations to eager load.
      *
      * @param QueryBuilder $queryBuilder
      * @param string       $resourceClass
      */
-    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass)
+    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, string $originAlias = 'o', string &$relationAlias = 'a')
```

Comment:
We need a central point of reference for the joins already made, so that there will be no repeated joins made by `EagerLoadingExtension` and `SearchFilter`. And I don't see why we should have default values for these parameters?